src/main/java/com/example/springbootbookshop/controller/BookController.java

```diff
@@ -0,0 +1,57 @@
+package com.example.springbootbookshop.controller;
+
+import com.example.springbootbookshop.dto.BookDto;
+import com.example.springbootbookshop.dto.CreateBookRequestDto;
+import com.example.springbootbookshop.entity.Book;
+import com.example.springbootbookshop.service.BookService;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @DeleteMapping(value = "/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteBookById(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @GetMapping(value = "/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public BookDto getBookById(@PathVariable Long id) {
```

Comment:
just FYI

you may also validate such things as @PathVariable, @RequestParams etc. It looks like:
```
@PathVariable @Positive Long id
```

In order to be able to perform such validation you need to annotate your controller class with @Validated annotation

again, this is just FYI - apply it or not :)